lib/core/src/Cardano/Wallet/Primitive/CoinSelection.hs

```diff
@@ -48,6 +48,8 @@ data CoinSelection = CoinSelection
       -- ^ Picked outputs
     , change  :: [Coin]
       -- ^ Resulting changes
+    , reserve :: Maybe Coin
```

Comment:
it can be only "one" ? not list?

Reply:
See comments above. In theory, we could also have deposits here. So I am thinking about having a list of `Reserve` where `Reserve = Withdrawal | Deposit`

Reply:
Will Withdraw and Deposit, if introduced will change on this level? Will there be different conditions in `rebalanceSelection` for them? for example you will deplete and transform Withdraw at first, and only then Deposit?

Reply:
No, they'll have the exact same effect hence why I initially went for a simple `Maybe Coin`. Perhaps it's not worth abstracting further here, in the end, both the deposit and withdrawal can be represent as a sum for this reserve. 
The difference lies in how the transaction is then constructed / signed / serialized. Withdrawals require a special entry as .. "withdrawals" in the transaction, as well as a special witness. 

Whereas deposit are implicit, when de-registering a stake key.